From 5a8cecb7d985afcb934fb577c1fbcced15d289a3 Mon Sep 17 00:00:00 2001
From: Dimitar Dimitrov <dimitar@dinux.eu>
Date: Sun, 15 Sep 2019 14:14:57 +0300
Subject: [PATCH 1/5] Fix fall through warning

Signed-off-by: Dimitar Dimitrov <dimitar@dinux.eu>
---
 gcc/config/pru/pru.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config/pru/pru.c b/gcc/config/pru/pru.c
index 416399ea5fd..8fea2a4b729 100644
--- a/gcc/config/pru/pru.c
+++ b/gcc/config/pru/pru.c
@@ -1650,7 +1650,7 @@ pru_print_operand (FILE *file, rtx op, int letter)
 	  return;
 	case 'Q':
 	  cond = swap_condition (cond);
-	  /* Fall through to reverse.  */
+	  /* Fall through.  */
 	case 'R':
 	  fprintf (file, "%s", pru_comparison_str (reverse_condition (cond)));
 	  return;
-- 
2.20.1

