From faf2828c80ac20368f9dbf06a17ca5827a55c79c Mon Sep 17 00:00:00 2001
From: Dimitar Dimitrov <dinuxbg@gmail.com>
Date: Sat, 13 Feb 2016 14:52:57 +0200
Subject: [PATCH 3/5] URGENT: Fix additional register name typo

Signed-off-by: Dimitar Dimitrov <dinuxbg@gmail.com>
---
 gcc/config/pru/pru.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config/pru/pru.h b/gcc/config/pru/pru.h
index f3b64d4..5ed3f09 100644
--- a/gcc/config/pru/pru.h
+++ b/gcc/config/pru/pru.h
@@ -337,7 +337,7 @@ typedef struct pru_args
 #define ADDITIONAL_REGISTER_NAMES       \
 {                                       \
   {"r2", 2},                          	\
-  {"r3", 2},                          	\
+  {"r3", 3},                          	\
   {"r4", 4},                          	\
 }
 
-- 
2.7.0

