From 0b2b6e3b4ba9628a6dd77f231858af8eded5f648 Mon Sep 17 00:00:00 2001
From: Dimitar Dimitrov <dinuxbg@gmail.com>
Date: Tue, 17 Nov 2015 23:19:41 +0200
Subject: [PATCH] Add libgloss definitions for pru target

Signed-off-by: Dimitar Dimitrov <dinuxbg@gmail.com>
---
 configure    | 3 +++
 configure.ac | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/configure b/configure
index 0dc0d1d..d444ffb 100755
--- a/configure
+++ b/configure
@@ -3617,6 +3617,9 @@ case "${target}" in
   powerpc*-*-*)
     libgloss_dir=rs6000
     ;;
+  pru-*-*)
+    libgloss_dir=pru
+    ;;
   sparc*-*-*)
     libgloss_dir=sparc
     ;;
diff --git a/configure.ac b/configure.ac
index 3f86c80..2f1852b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -954,6 +954,9 @@ case "${target}" in
   powerpc*-*-*)
     libgloss_dir=rs6000
     ;;
+  pru-*-*)
+    libgloss_dir=pru
+    ;;
   sparc*-*-*)
     libgloss_dir=sparc
     ;;
-- 
2.6.2

