From 55fdec33ccf70103ef96e0868b8c510040fa9bdc Mon Sep 17 00:00:00 2001
From: Dimitar Dimitrov <dimitar@dinux.eu>
Date: Sun, 30 Apr 2017 23:40:29 +0300
Subject: [PATCH 4/6] WIP: New reg constraint format

Signed-off-by: Dimitar Dimitrov <dimitar@dinux.eu>
---
 libgloss/pru/include/pru/io.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/libgloss/pru/include/pru/io.h b/libgloss/pru/include/pru/io.h
index 945ff06f3..2976e3649 100644
--- a/libgloss/pru/include/pru/io.h
+++ b/libgloss/pru/include/pru/io.h
@@ -30,25 +30,25 @@
 
 static inline void write_r30(unsigned int val)
 {
-	asm volatile ("mov\tr30, %0" : : "r"(val));
+	asm volatile ("mov\tr30, %x0" : : "r"(val));
 }
 
 static inline unsigned int read_r30(void)
 {
 	unsigned int val;
-	asm volatile ("mov\t%0, r30" : "=r"(val) :);
+	asm volatile ("mov\t%x0, r30" : "=r"(val) :);
 	return val;
 }
 
 static inline void write_r31(unsigned int val)
 {
-	asm volatile ("mov\tr31, %0" : : "r"(val));
+	asm volatile ("mov\tr31, %x0" : : "r"(val));
 }
 
 static inline unsigned int read_r31(void)
 {
 	unsigned int val;
-	asm volatile ("mov\t%0, r31" : "=r"(val) :);
+	asm volatile ("mov\t%x0, r31" : "=r"(val) :);
 	return val;
 }
 
-- 
2.11.0

