From d9a72afa740ae200dbc28735adb8af9c90dc0e46 Mon Sep 17 00:00:00 2001
From: Dimitar Dimitrov <dimitar@dinux.eu>
Date: Sat, 6 May 2017 23:30:05 +0300
Subject: [PATCH 5/6] Add support for argc/argv

Reference: SPRUHV7A, section 2.6

Signed-off-by: Dimitar Dimitrov <dimitar@dinux.eu>
---
 libgloss/pru/crt0.S | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/libgloss/pru/crt0.S b/libgloss/pru/crt0.S
index d0ffc5f92..860428351 100644
--- a/libgloss/pru/crt0.S
+++ b/libgloss/pru/crt0.S
@@ -47,6 +47,15 @@ ctors_done:
 	/* Just in case main() tries to access argc, argv[] and envp. */
 	zero	r14, 3 * 4
 
+#if !defined(MINRT)
+	.weak __c_args__
+	ldi32	r5, __c_args__
+	qbeq	__skip_c_args, r5, 0
+	lbbo	r14, r5, 0, 4	/* argc */
+	add	r15, r5, 4	/* argv */
+__skip_c_args:
+#endif
+
 	/* Call main */
 	call	main
 
-- 
2.11.0

